cpp/src/io/parquet/page_string_decode.cu

```diff
@@ -625,8 +655,12 @@ __global__ void __launch_bounds__(preprocess_block_size) gpuComputePageStringSiz
       break;
     case Encoding::PLAIN:
       dict_size = static_cast<int32_t>(end - data);
-      str_bytes = is_bounds_pg ? totalPlainEntriesSize(data, dict_size, start_value, end_value)
-                               : dict_size - sizeof(int) * (pp->num_input_values - pp->num_nulls);
+      if ((s->col.data_type & 7) == FIXED_LEN_BYTE_ARRAY) {
+        str_bytes = totalFixedLenByteArraySize(s->dtype_len_in, dict_size, start_value, end_value);
```

Comment:
or just get rid of `totalFixedLenByteArraySize` and calculate `str_bytes` directly